src/v/raft/tests/raft_fixture.cc

```diff
@@ -60,7 +60,7 @@ ss::future<> channel::stop() {
         auto f = _gate.close();
 
         for (auto& m : _messages) {
-            m.resp_data.set_exception(ss::abort_requested_exception());
+            m.resp_data.maybe_set_exception(ss::abort_requested_exception());
```

Comment:
what exactly is doing a double set? this seems. a bit hacky, wondering if it is masking some other problem.

Reply:
As far as I understand it's a race between `stop()` and `dispatch_loop()`. I only saw it once though, and I failed to reproduce it. I've changed it to less hacky.